src/python/WMCore/MicroService/Unified/Common.py

```diff
@@ -101,10 +128,43 @@ def phedexValidBlocks(datasets, phedexUrl):
         rows = json.loads(row['data'])
         dsetBlockSize.setdefault(dataset, {})  # flat dict in the format of blockName: blockSize
         for item in rows['phedex']['block']:
-            dsetBlockSize[dataset].update({item['name']: item['bytes']})
+            block = {item['name']: {'blockSize': item['bytes'], 'locations': []}}
```

Comment:
again, phedex specific part, you may consider write it in more generic way to handle both phedex and rucio streams of data.

Reply:
Yes, the rucio part will be made in a separate PR because this one has already too many changes. I'm planning to make separate functions for Rucio calls (code is pretty small anyways)